File: src/main/java/com/hubsante/model/edxl/DistributionStatus.java
package com.hubsante.model.edxl;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

public enum DistributionStatus {
    ACTUAL("Actual"), EXERCISE("Exercise"), SYSTEM("System"), TEST("Test"), UNKOWN("Unknown"), NO_APPROPRIATE_DEFAULT("NoAppropriateDefault");

    private String value;
    DistributionStatus(String value) { this.value = value; }

    @JsonValue
    public String getValue() {
        return value;
    }

    @JsonCreator
    public static DistributionStatus fromValue(String value) {
        for (DistributionStatus e : DistributionStatus.values()) {
            if (e.value.equals(value)) {
                return e;
            }
        }
        throw new IllegalArgumentException("Unexpected value '" + value + "'");
    }

    @Override
    public String toString() {
        return String.valueOf(value);
    }
}

File: generator_php-symfony/package/src/Model/edxl/DistributionStatus.php
<?php

/**
 * Copyright © 2023-2025 Agence du Numerique en Sante (ANS)
 * Licensed under the Apache License, Version 2.0
 */

namespace App\Model\edxl;

// use JMS\Serializer\Annotation as JMS;
// use Symfony\Component\Validator\Constraints as Assert;

enum DistributionStatus: string
{
    case ACTUAL = 'Actual';
    case EXERCISE = 'Exercise';

    // /**
    //  * @Assert\NotBlank
    //  * @JMS\SerializedName("distributionStatus")
    //  * @JMS\Type("string")
    //  */

    public function getValue(): string
    {
        return $this->value;
    }

    public static function fromValue(string $value): self
    {
        foreach (self::cases() as $case) {
            if ($case->value === $value) {
                return $case;
            }
        }

        throw new \InvalidArgumentException("Unexpected value '$value'");
    }

    public function __toString(): string
    {
        return $this->value;
    }
}
